datafusion/physical-plan/src/aggregates/group_values/row.rs

```diff
@@ -86,15 +102,42 @@ impl GroupValuesRows {
             group_values: None,
             hashes_buffer: Default::default(),
             random_state: Default::default(),
+            var_map: ArrowBytesMap::new(OutputType::Utf8),
+            num_groups: 0,
         })
     }
 }
 
 impl GroupValues for GroupValuesRows {
     fn intern(&mut self, cols: &[ArrayRef], groups: &mut Vec<usize>) -> Result<()> {
+        let cols_var = &cols[1];
+        groups.clear();
+        self.var_map.insert_if_new(
+            cols_var,
+            // called for each new group
+            |_value| {
+                // assign new group index on each insert
+                let group_idx = self.num_groups;
+                self.num_groups += 1;
+                group_idx
+            },
+            // called for each group
+            |group_idx| {
+                groups.push(group_idx);
+            },
+        );
+
+
+        let u64_vec: Vec<u64> = groups.iter().map(|&x| x as u64).collect();
```

Comment:
It might be faster to write into `u64_vec` directly rather than write to groups and then copy over